File: 749-shortest-completing-word/shortest-completing-word.cpp
class Solution {
public:
    string shortestCompletingWord(string licensePlate, vector<string>& words) {
        unordered_map<char,int> k;
        for(auto i : licensePlate){
            if(isalpha(i)){
                char ch = tolower(i); 
                k[ch]++;
            }
        }
        int mi = INT_MAX;
        string res = "";
        
        for(auto j : words){  
            unordered_map<char,int> d;
            for(auto h : j){
                d[h]++;
            }
            
            bool s = true;  
            for(auto p : k){
                char a = p.first;
                int b = p.second;
                if(d.count(a) != 0){
                    if(d[a] < b){
                        s = false;
                        break;
                    }
                }
                else{
                    s = false;
                    break;
                }
            }
            
            if(s && j.length() < mi){ 
                res = j;
                mi = j.size();
            }
        }  
        
        return res;
    }
};

File: 749-shortest-completing-word/shortest-completing-word.py
class Solution:
    def shortestCompletingWord(self, licensePlate: str, words: List[str]) -> str:
        k={}
        for i in licensePlate:
            if(i.isalpha()):
                i = i.lower()
                if(i in k):
                    k[i]+=1
                else:
                    k[i]=1
        
        mi = float('inf')
        res = None
        for j in words:
            d={}
            for c in j:
                if(c in d):
                    d[c]+=1
                else:
                    d[c]=1
            h = True
            for a,b in k.items():
                if a in d:
                    if(d[a]<b):
                        h = False
                        break
                else:
                    h = False
                    break
            if(h and len(j)<mi):
                res = j
                mi = len(j)
        return res

